File: 2021-06-27/app/src/main/java/com/github/skiiyis/hotfix/MainActivity.java
package com.github.skiiyis.hotfix;

import android.app.Activity;
import android.os.Bundle;
import android.widget.TextView;

import com.skiiyis.study.hotfix.R;

public class MainActivity extends Activity {

    // public static HotFixProxy __change;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        checkPatch();
        ((TextView) findViewById(R.id.text)).setText(problemMethod());
        ((TextView) findViewById(R.id.text)).setOnClickListener(v -> ((TextView) findViewById(R.id.text)).setText(problemMethod()));
    }

    private void checkPatch() {
        HotFixCheckUtil.checkAndHotFix(this);
    }

    public String noProblemMethod() {
        return "I'm no problem method!!";
    }

    private String problemMethod() {
        return "I'm problem method!!";
    }
}

File: 2021-06-27/app/src/main/java/com/skiiyis/study/hotfix/MainActivity.kt
package com.skiiyis.study.hotfix

import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        findViewById<TextView>(R.id.text).text = problemMethod()
        checkPatch()
    }

    private fun checkPatch() {
        HotFixCheckUtil.checkAndHotFix(this)
    }

    private fun problemMethod(): String {
        return "I'm problem method!!"
    }

}